test/GovPool.test.js

```diff
@@ -1744,5 +1793,180 @@ describe("GovPool", () => {
+          await govPool.delegate(micropool, wei("50000000000000000000"), [30, 31], { from: delegator3 });
+
+          await govPool.voteDelegated(1, wei("250000000000000000000"), [], { from: micropool });
+
+          await setTime((await getCurrentBlockTime()) + 10000);
+
+          await govPool.moveProposalToValidators(1);
+
+          await validators.vote(1, wei("100"), false);
+          await validators.vote(1, wei("1000000000000"), false, { from: SECOND });
+
+          await govPool.execute(1);
+          await govPool.claimRewards([1], { from: micropool });
+
+          await govPool.undelegate(micropool, wei("100000000000000000000"), [], { from: delegator1 });
+          await govPool.undelegate(micropool, wei("100000000000000000000"), [], { from: delegator2 });
+          await govPool.undelegate(micropool, wei("50000000000000000000"), [], { from: delegator3 });
+
+          const micropoolBalance = await rewardToken.balanceOf(micropool);
+          const balance1 = await rewardToken.balanceOf(delegator1);
+          const balance2 = await rewardToken.balanceOf(delegator2);
+          const balance3 = await rewardToken.balanceOf(delegator3);
+
+          assertNoZerosBalanceDistribution([balance1, balance2, balance3, micropoolBalance], [32, 32, 16, 20]);
+        });
+
+        it("should give the proper rewards with multiple async delegates", async () => {
+          await govPool.delegate(micropool, wei("1000"), [10, 11, 12, 13], { from: delegator1 });
+          await govPool.voteDelegated(1, wei("800"), [], { from: micropool });
+
+          await govPool.delegate(micropool, wei("1000"), [20, 21, 22, 23], { from: delegator2 });
+          await govPool.voteDelegated(1, wei("800"), [], { from: micropool });
+
+          await govPool.delegate(micropool, wei("500"), [30, 31], { from: delegator3 });
+          await govPool.voteDelegated(1, wei("800"), [], { from: micropool });
+
+          await setTime((await getCurrentBlockTime()) + 10000);
+
+          await govPool.undelegate(micropool, wei("1000"), [10, 11, 12, 13], { from: delegator1 });
+          await govPool.undelegate(micropool, wei("1000"), [20, 21, 22, 23], { from: delegator2 });
+          await govPool.undelegate(micropool, wei("500"), [30, 31], { from: delegator3 });
+
+          const balance1 = await rewardToken.balanceOf(delegator1);
+          const balance2 = await rewardToken.balanceOf(delegator2);
+          const balance3 = await rewardToken.balanceOf(delegator3);
+
+          assertNoZerosBalanceDistribution([balance1, balance2, balance3], [19, 9, 2]);
+        });
+      });
+    });
+  });
+
+  describe("ERC721Power", () => {
+    let POOL_PARAMETERS;
+
+    beforeEach("setup", async () => {
+      POOL_PARAMETERS = await getPoolParameters(nftPower.address);
+
+      await deployPool(POOL_PARAMETERS);
+      await setupTokens();
+    });
+
+    describe("staking", () => {
```

Comment:
Could you please add a test when the same user delegates twice?

Reply:
done